frontend/src/views/UsersList.js

```diff
@@ -0,0 +1,227 @@
+      return (
+        <li key={user['id']} className={styles['user-list-item']}>
+          <div className={styles['user-fullname']}>
+            <Link
+              className={styles['user-fullname-link']}
+              to={'/figures/user/' + user['id']}
+            >
+              {user['fullname']}
+            </Link>
+          </div>
+          <div className={styles['username']}>
+            {user['username']}
+          </div>
+          <div className={styles['is-active']}>
+            {user['is_active'] ? <FontAwesomeIcon icon={faCheck} className={styles['checkmark-icon']} /> : '-'}
+          </div>
+          <div className={styles['date-joined']}>
+            {user['date_joined']}
+          </div>
+          <div className={styles['number-of-courses']}>
+            {user['courses'].length}
+          </div>
+          <div className={styles['action-container']}>
+            <Link
+              className={styles['user-action']}
+              to={'/figures/user/' + user['id']}
+            >
+              User details
+            </Link>
+          </div>
+        </li>
+      )
+    })
+
+    return (
+      <div className="ef--layout-root">
+        <HeaderAreaLayout>
+          <HeaderContentStatic
+            title='Users list'
+            subtitle={'This view allows you to browse your sites users. Total number of results: ' + this.state.count + '.'}
+          />
+        </HeaderAreaLayout>
+        <div className={cx({ 'container': true, 'users-content': true})}>
+          <ListSearch
+            valueChangeFunction={this.setSearchQuery}
+            inputPlaceholder='Search by users name, username or email...'
+          />
+          {this.state.pages ? (
+            <Paginator
+              pageSwitchFunction={this.getUsers}
+              currentPage={this.state.currentPage}
+              perPage={this.state.perPage}
+              pages={this.state.pages}
+              changePerPageFunction={this.setPerPage}
+            />
+          ) : ''}
+          <ul className={styles['users-list']}>
+            <li key='list-header' className={cx(styles['user-list-item'], styles['list-header'])}>
+              <div className={styles['user-fullname']}>
+                <button
+                  className={styles['sorting-header-button']}
+                  onClick={() => (this.state.ordering !== 'profile__name') ? this.setOrdering('profile__name') : this.setOrdering('-profile__name')}
+                >
+                  <span>
+                    User full name
+                  </span>
+                  {(this.state.ordering === 'profile__name') ? (
+                    <FontAwesomeIcon icon={faAngleDoubleUp} />
+                  ) : (this.state.ordering === '-profile__name') ? (
+                    <FontAwesomeIcon icon={faAngleDoubleDown} />
+                  ) : ''}
+                </button>
+              </div>
+              <div className={styles['username']}>
+                <button
```

Comment:
@grozdanowski Not for this PR, but would these button elements be a good candidate for abstraction to a 'Button' class?

Reply:
Yup - good idea. Also yeah, not for this PR since it needs to be done in a few places.